### PR TITLE
Add NHHall reverb UGen

### PR DESCRIPTION
**Pull request by nhthn**
a [new algorithmic reverb](https://github.com/snappizz/nh-ugens) i'm developing. it has low CPU consumption, true stereo signal path, infinite hold support, a handful of EQ and diffusion parameters to mess with, and a decent sound if i may say so myself. it's also MIT licensed.

as the helpfile warning states, this UGen will probably see a few minor tweaks to its sound in the near future. i still encourage people to use it since these changes won't be drastic and are unlikely to upheave a composition (mostly fiddling with delay line lengths to neutralize artifacts).

### CODE REVIEW

**Comment by telephon** [comment]
would it be hard to make it multichannel?

**Comment by joslloand** [comment]
> best architecture depends on the flavor of multichannel -- ambisonics? 5.1 surround?

Obviously, Ambisonics is best! ;-)

File: source/CMakeLists.txt [hunk 1]
```diff
@@ -243,7 +244,7 @@ endif()
 # OteyPianoUGens
 BUILD_PLUGIN(OteyPianoUGens "OteyPianoUGens/OteyPiano.cpp;OteyPianoUGens/dwgs.cpp;OteyPianoUGens/filter.cpp;OteyPianoUGens/hammer.cpp;OteyPianoUGens/reverb.cpp"
              "" "OteyPianoUGens")
-             
+
```
Comment (brianlheim) [inline source/CMakeLists.txt]: could you please revert these whitespace changes? IIRC might conflict with #146 if merged in this state.

Comment (snappizz) [reply source/CMakeLists.txt]: :ok_hand:

**Comment by patrickdupuis** [comment]
@snappizz What's the status of this PR? Are we ready to merge?

**Comment by telephon** [comment]
@snappizz sorry for the silence, I had overlooked your question.

>would you be more interested in a mono-to-multichannel or a multichannel-to-multichannel reverb?

I thought that stereo input is rather particular (but OK!).  Maybe the UGen should be called `NHHall2` to indicate this.

If you can separate subsets of your implementation as separate UGens, then one can just build one's own multichannel version in sclang, maybe?

File: source/NHUGens/sc/NHHall.sc [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            hiFreq,
+            hiRatio,
+            earlyDiffusion,
+            lateDiffusion,
+            modRate,
+            modDepth
+        );
+    }
+
+    init { arg ... theInputs;
```
Comment (telephon) [inline source/NHUGens/sc/NHHall.sc]: better use `|...theInputs|` when you use them above, too.

**Comment by snappizz** [comment]
the only serious issue with this UGen is a denormal number bug with ARM processors (rpi, bela, etc.). i've been working with patrick to try to track it down.

other than that, i'm dissatisfied with a weird stereo "wobble" in the initial onset of the reverb tail, so some tweaks will be necessary. both of these issues are documented in the helpfile, so this should be good to go.

> I thought that stereo input is rather particular (but OK!). Maybe the UGen should be called NHHall2 to indicate this.

thanks for the suggestion, but i think the name is fine. the vast majority of commercial reverbs are stereo as well.

File: source/NHUGens/sc/NHHall.sc (hunk 2 above)
Comment (snappizz) [reply source/NHUGens/sc/NHHall.sc]: :ok_hand:

**Pull request merged**